docs/openapi/schemas.yaml

```diff
@@ -1598,3 +1598,381 @@ ApiKeyResponse:
+      $ref: '#/Id'
+    type:
+      $ref: '#/SuggestionType'
+    rank:
+      description: Type-specific numeric rank value (e.g., "domainTraffic" for a broken-backlink) helps sorting
+      type: number
+      example: 45600
+    data:
+      description: Map containing type-specific details of what fix is suggested
+      type: object
+      additionalProperties: true
+      oneOf:
+        - $ref: '#/BrokenBacklinksRedirectData'
+    kpiDeltas:
+      description: Map containing metrics estimated to experience lifts or losses due to the issue
+      type: object
+      additionalProperties: true
+      oneOf:
+        - $ref: '#/BrokenBacklinksKpiDeltas'
+    status:
+      description: Status of this suggestion; status reflects overall fix execution, flagged here for optimization
+      $ref: '#/SuggestionStatus'
+    createdAt:
+      description: UTC timestamp of when the suggestion was created
+      $ref: '#/DateTime'
+    updatedAt:
+      description: UTC timestamp of when the suggestion was last updated
+      $ref: '#/DateTime'
+  required:
+    - id
+    - opportunityId
+    - type
+    - rank
+    - data
+    - status
+    - createdAt
+    - updatedAt
+SuggestionList:
+  type: array
+  items:
+    $ref: '#/Suggestion'
+SuggestionCreate:
+  type: object
+  properties:
+    opportunityId:
+      description: UUID of the opportunity this suggestion belongs to
+      $ref: '#/Id'
+    type:
+      $ref: '#/SuggestionType'
+    rank:
+      description: Type-specific numeric rank value (e.g., "domainTraffic" for a broken-backlink) helps sorting
+      type: number
+      example: 45600
+    data:
+      description: Map containing type-specific details of what fix is suggested
+      type: object
+      additionalProperties: true
+      oneOf:
+        - $ref: '#/BrokenBacklinksRedirectData'
+    kpiDeltas:
+      description: Map containing metrics estimated to experience lifts or losses due to the issue
+      type: object
+      additionalProperties: true
+      oneOf:
+        - $ref: '#/BrokenBacklinksKpiDeltas'
+  required:
+    - opportunityId
+    - type
+    - rank
+    - data
+SuggestionCreateList:
+  type: array
+  items:
+    $ref: '#/SuggestionCreate'
+SuggestionUpdate:
+  type: object
+  properties:
+    rank:
+      description: Type-specific numeric rank value (e.g., "domainTraffic" for a broken-backlink) helps sorting
+      type: number
+      example: 12300
+    data:
+      description: Map containing type-specific details of what fix is suggested
+      type: object
+      additionalProperties: true
+      oneOf:
+        - $ref: '#/BrokenBacklinksRedirectData'
+    kpiDeltas:
+      description: Map containing metrics estimated to experience lifts or losses due to the issue
+      type: object
+      additionalProperties: true
+      oneOf:
+        - $ref: '#/BrokenBacklinksKpiDeltas'
+SuggestionStatusUpdate:
+  type: array
+  items:
+    type: object
+    properties:
+      id:
```

Comment:
This id is also lost :) 
<img width="1173" alt="Screenshot 2024-11-15 at 19 10 03" src="https://github.com/user-attachments/assets/1e5c776b-2e25-43a6-9b90-173f14ce75fc">

Reply:
Found. The referenced Id schema had readOnly: true, so the property wasn't rendered in request body. We needed to explicitly set readOnly: false for the property in the request body schema.